src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return iteratorByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldValue = iteratorByIndex(index).value;
+        iteratorByIndex(index).value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removedNode = iteratorByIndex(index);
+        removeNode(removedNode);
+        return removedNode.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> removedNode = iteratorByValue(object);
+        if (removedNode == null) {
+            return false;
+        }
+        removeNode(removedNode);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> iteratorByIndex(int index) {
+        if (isEmpty() || index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index [" + index + "] out of bounds!");
+        }
+        Node<T> tmp = head;
+        for (int indexCount = 0; indexCount < size; indexCount++) {
+
+            if (indexCount == index) {
+                return tmp;
+            }
+            tmp = tmp.next;
+        }
+        return null;
+    }
+
+    private Node<T> iteratorByValue(T value) {
+        if (isEmpty()) {
+            throw new NullPointerException("LinkedList is empty");
+        }
+        Node<T> tmp = head;
+        for (int indexCount = 0; indexCount < size; indexCount++) {
+            if (value == null) {
+                if (value == tmp.value) {
+                    return tmp;
+                } else {
+                    tmp = tmp.next;
+                }
+            } else {
+                if (value.equals(tmp.value)) {
+                    return tmp;
+                } else {
+                    tmp = tmp.next;
+                }
+            }
+        }
+        return null;
+    }
+
+    private void removeNode(Node<T> removedNode) {
+        if (removedNode == head && removedNode == tail) {
+            removedNode = null;
+            head = null;
+            tail = null;
+            size--;
+        } else if (removedNode == head) {
+            removedNode.next.prev = null;
+            head = removedNode.next;
+            size--;
+        } else if (removedNode == tail) {
+            removedNode.prev.next = null;
+            tail = removedNode.prev;
+            size--;
+        } else {
+            removedNode.prev.next = removedNode.next;
+            removedNode.next.prev = removedNode.prev;
+            size--;
+        }
+    }
```

Comment:
it should be method ulink (see [common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#create-a-separate-method-unlinknode-node))